src/main/java/fact/datacorrection/DrsCalibration.java

```diff
@@ -108,52 +119,67 @@ public Data process(Data data) {
-            log.error(" data .fits file did not contain the value for the key "
-                    + key + ". cannot apply drscalibration");
-            throw new RuntimeException(
-                    " data .fits file did not contain the value for the key \"" + key + "\". Cannot apply drs calibration)");
-        }
+        if (!reverse) {
+            short[] rawData = (short[]) data.get(key);
+            if (rawData == null) {
+                log.error(" data .fits file did not contain the value for the key "
+                        + key + ". cannot apply drscalibration");
+                throw new RuntimeException(
+                        " data .fits file did not contain the value for the key \"" + key + "\". Cannot apply drs calibration)");
+            }
 
-        double[] rawfloatData = new double[rawData.length];
-        // System.arraycopy(rawData, 0, rawfloatData, 0, rawfloatData.length);
-        for (int i = 0; i < rawData.length; i++) {
-            rawfloatData[i] = rawData[i];
+            rawfloatData = new double[rawData.length];
+            // System.arraycopy(rawData, 0, rawfloatData, 0, rawfloatData.length);
+            for (int i = 0; i < rawData.length; i++) {
+                rawfloatData[i] = rawData[i];
+            }
+        } else {
+            Utils.isKeyValid(data, key, double[].class);
+            rawfloatData = (double[]) data.get(key);
         }
 
-        short[] startCell = (short[]) data.get("StartCellData");
+        short[] startCell = (short[]) data.get(startCellKey);
         if (startCell == null) {
             log.error(" data .fits file did not contain startcell data. cannot apply drscalibration");
             return null;
         }
-        log.debug("raw data has {} elements", rawData.length);
+        log.debug("raw data has {} elements", rawfloatData.length);
         log.debug("StartCellData has {} elements", startCell.length);
 
         double[] output = rawfloatData;
         if (!key.equals(outputKey)) {
-            output = new double[rawData.length];
+            output = new double[rawfloatData.length];
         }
 
-        double[] calibrated = applyDrsCalibration(rawfloatData, output, startCell);
-
-        data.put(outputKey, calibrated);
+        double[] calibrated = applyDrsCalibration(rawfloatData, output, startCell, reverse);
+        if (!reverse)
+            data.put(outputKey, calibrated);
+        else {
+            short[] decalibratedShortData = new short[calibrated.length];
+            // System.arraycopy(rawData, 0, rawfloatData, 0, rawfloatData.length);
+            for (int i = 0; i < calibrated.length; i++) {
+                decalibratedShortData[i] = (short) calibrated[i];
+            }
+            data.put(outputKey, decalibratedShortData);
+        }
 
-        // add color value if set
+        //TODO add color value if set
 
         return data;
     }
 
     public double[] applyDrsCalibration(double[] data, double[] destination,
```

Comment:
I still think two functions would be more clear. `applyDrsCalibration` and `revertDrsCalibration`.

Reply:
split into two functions